voltdb/src/main/java/com/yahoo/ycsb/db/voltdb/ConnectionHelper.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2015-2019 YCSB contributors. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */ 
+
+/*
+ * Manage a mapping of one VoltDB client to every 50 YCSB client threads.
+ */
+package com.yahoo.ycsb.db.voltdb;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.util.concurrent.CountDownLatch;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.voltdb.client.Client;
+import org.voltdb.client.ClientConfig;
+import org.voltdb.client.ClientFactory;
+
+/**
+ * Help class to create VoltDB connections for YCSB benchmark.
+ */
+public final class ConnectionHelper {
+  
+  /**
+   * Default port for VoltDB.
+   */
+  private static final int VOLTDB_DEFAULT_PORT = 21212;
+
+  /**
+   * hidden constructor.
+   */
+  private ConnectionHelper() {
+
+  }
+
+  /**
+   * Creates a factory used to connect to a VoltDB instance. (Note that if a
+   * corresponding connection exists, all parameters other than 'servers' are
+   * ignored)
+   * 
+   * @param clientId  A unique identifier for the connecting client
+   * @param servers   The comma separated list of VoltDB servers in
+   *                  hostname[:port] format that the instance will use.
+   * @param user      The username for the connection
+   * @param password  The password for the specified user
+   * @param ratelimit A limit on the number of transactions per second for the
+   *                  VoltDB instance
+   * @return The existing factory if a corresponding connection has already been
+   *         created; the newly created one otherwise.
+   * @throws IOException          Throws if a connection is already open with a
+   *                              different server string.
+   * @throws InterruptedException
+   */
+  public static synchronized Client createConnection(Long clientId, String servers, String user, String password,
```

Comment:
Does this still need to be synchronized?

Reply:
No. Fixed.